sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1553,6 +1558,7 @@ def __init__(self,
                      cluster_name: str,
                      cluster_yaml: str,
                      head_ip: Optional[str] = None,
```

Comment:
This head_ip can be stale. How about we have a `@property` function called `head_ip` instead
```
@property
def head_ip(self):
    return self.stable_ips[0][1]
```

This needs to be careful due to the backward compatibility, for loading existing handles from the our cluster database.

Reply:
When would this `heap_ip` value be stale? From what I see, it's updated and invalidated at the same places as the stable ip list currently.

Reply:
Oh, I was trying to say that if we are going to cache the external IP as well, the `head_ip` is a duplicated variable that may not be needed.

Reply:
That makes sense!